src/hooks/useFirestoreConnect.ts

```diff
@@ -15,23 +17,36 @@ export type Defined<T> = T & Exclude<T, undefined>;
  * This type allows us to automagically constrain the storeAs
  * parameter to keys that exist within ValidFirestoreKeys, ensuring
  * that we can't forget to define it in our types when using functions
- * that rely on this type (eg. useSparkleFirestoreConnect)
+ * that rely on this type (eg. useFirestoreConnect)
  *
  * @see ValidFirestoreKeys
- * @see useSparkleFirestoreConnect
+ * @see useFirestoreConnect
  * @see ReduxFirestoreQuerySetting
  */
-export interface SparkleRFQConfig extends ReduxFirestoreQuerySetting {
+
+export type SparkleUseFirestoreConfigObject = ReduxFirestoreQuerySetting & {
   storeAs?: ValidFirestoreKeys;
-}
+};
+
+/**
+ * This type lists all allowed argument types for @useFirestoreConnect function
+ *
+ * @see SparkleUseFirestoreConfigObject
+ * @see ValidFirestoreKeys
+ */
+export type SparkleRFQConfig =
+  | SparkleUseFirestoreConfigObject
+  | SparkleUseFirestoreConfigObject[]
+  | ValidFirestoreKeys
+  | undefined;
```

Comment:
https://github.com/sparkletown/sparkle/pull/801#discussion_r557260510

Reply:
I fixed this up in my PR, so no longer an issue, but I feel like using `undefined` in a grouping type like this will likely cause more issues/confusion, so would be best to avoid it in future code.

We are still able to pass the `undefined` queries to the `useFirestoreConnect` function because of the `config?` 'optional' syntax.